Add retries in verification script and provisioning tests

maelk opened the pull request:
Avoid failing without retrying to wait for the dev env
to initialize properly. Add verify target in Makefile all
Add a provisioning test script that verifies the provisioning and deprovisioning of the machines

maelk committed:
Add retries in verification script

Avoid failing without retrying to wait for the dev env
to initialize properly. Add verify target in Makefile all

nordixinfra commented:
Can one of the admins verify this patch?

namnx228 commented on 04_verify.sh:
```diff
@@ -41,67 +76,94 @@ differs(){
-
-      # Verify credentials exist
-      CRED_NAME="$(echo "${BM_HOST}" | jq -r '.spec.bmc.credentialsName')"
-      CRED_SECRET="$(kubectl get secret "${CRED_NAME}" -n metal3 -o json | jq '.data')"
```
in the case we use kinder, I think we may need to export KUBECONFIG
```sh
export KUBECONFIG="kinder get kubeconfig-path"
```

maelk replied:
ok, I'll add a Kubeconfig variable

maelk committed:
Add a test script

the script provisions the machines and tests the setup. It will
also deprovision and test

kanwar-saad commented on 05_test.sh:
```diff
@@ -0,0 +1,241 @@
+
+
+# provision the machines
+for name in centos-0 centos-1; do
```
you can pre define this array .. it is used in all loops later

kanwar-saad commented on 05_test.sh:
```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env bash
```
since it is a new file .. you can keep it free of linter errors.

kanwar-saad commented on lib/common.sh:
```diff
@@ -124,3 +132,88 @@ function list_nodes() {
        | sed 's/"//g'
 }
+
+iterate(){
```
missing function doc ... document description, inputs and outputs.

kanwar-saad commented on lib/common.sh:
```diff
@@ -124,3 +132,88 @@ function list_nodes() {
+}
+
+
+# Check the return code and log
```
same for rest of the functions

maelk committed:
Refactor 04_verify.sh and 05_test.sh bash code
